### PR TITLE
minSubarraySizeWithDegree

### PR DESCRIPTION
**Pull request by AugustoQueiroz**
**Fixes issue:** #1784

**Changes:**
Created a function that returns the size of the smallest subarray with the same degree as the original array.
The function uses a two-pointers algorithm to sweep through the array, looking for the subarrays that satisfy the requirement, and then returns the size of the smallest one.
Time Complexity: O(n)

### CODE REVIEW

File: code/unclassified/minSubarraySizeWithDegree/minSubarraySizeWithDegree.cpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
+#include <bits/stdc++.h>
+#include <unordered_map>
+
+using namespace std;
```
Comment (Vitorvgc) [inline code/unclassified/minSubarraySizeWithDegree/minSubarraySizeWithDegree.cpp]: Don't forget to add the required comment as stated in the [contribution guidelines](https://github.com/OpenGenus/cosmos/wiki/contribute).

File: code/unclassified/minSubarraySizeWithDegree/minSubarraySizeWithDegree.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
+#include <bits/stdc++.h>
```
Comment (Vitorvgc) [inline code/unclassified/minSubarraySizeWithDegree/minSubarraySizeWithDegree.cpp]: Is is not preferable to use this header as it is not supported by all compilers.

File: code/unclassified/minSubarraySizeWithDegree/minSubarraySizeWithDegree.cpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+using namespace std;
+
+int minSubArraySizeWithDegree(const vector<int> nums) {
+	unordered_map <int, int> m;
+	priority_queue<int> pq;
+	int best_degree = 1;
+	int curr_best_degree = nums[0];
+	int i,
+		j,
+		s = nums.size();
```
Comment (Vitorvgc) [inline code/unclassified/minSubarraySizeWithDegree/minSubarraySizeWithDegree.cpp]: I think it is better to put these lines aligned or in the same line.

File: code/unclassified/minSubarraySizeWithDegree/minSubarraySizeWithDegree.cpp (hunk 2 above)
Comment (AugustoQueiroz) [reply code/unclassified/minSubarraySizeWithDegree/minSubarraySizeWithDegree.cpp]: Ok, makes sense

**Comment by AugustoQueiroz** [comment]
Your comments should be fixed now.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**